ledger/participant-integration-api/src/main/scala/platform/apiserver/services/transaction/ApiTransactionService.scala

```diff
@@ -137,13 +141,28 @@ private[apiserver] final class ApiTransactionService private (
       .map { case LfEventId(transactionId, _) =>
         lookUpTreeByTransactionId(TransactionId(transactionId), request.requestingParties)
       }
-      .getOrElse(
+      .getOrElse {
+        val msg = s"invalid eventId: ${request.eventId}"
         Future.failed(
-          Status.NOT_FOUND
-            .withDescription(s"invalid eventId: ${request.eventId}")
-            .asRuntimeException()
+          errorsVersionsSwitcher.choose(
+            v1 = Status.NOT_FOUND
+              .withDescription(msg)
+              .asRuntimeException(),
+            v2 = LedgerApiErrors.CommandValidation.InvalidArgument
```

Comment:
@tudor-da Please have a look

Reply:
Discussed offline. Looks better with the new method in the `LoggingTransactionErrorImpl`

Reply:
Discussed oflline. Looks good 👍